packages/driver/src/cy/commands/navigation.coffee

```diff
@@ -719,10 +733,17 @@ module.exports = (Commands, Cypress, cy, state, config) ->
             .then ->
               onLoad(resp)
           else
-            ## if we've already visited a new superDomain
+            ## if we've already visited a new origin
             ## then die else we'd be in a terrible endless loop
             if previousDomainVisited
-              return cannotVisit2ndDomain(remote.origin, previousDomainVisited, options._log)
+              reason = ''
+
+              if remote.protocol isnt existing.protocol
+                reason = 'differentProtocol'
+              if remote.port isnt existing.port
+                reason = 'differentPort'
```

Comment:
maybe something like this would get the same effect across, while still allowing for a specific reason...

```coffee
reasons = []
if remote.protocol isnt existing.protocol
	reasons.push('protocol')
if remote.port isnt existing.port
	reasons.push('port')
if remote.superdomain isnt existing.superdomain # if this works
	reasons.push('superdomain')
```

then, the error could just read "You've switched to a different origin (different protocol and port)" or whatever, so the user knows all the reasons why the origin is different, instead of just 1 arbitrary reason

Reply:
True